akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessImpl.scala

```diff
@@ -168,7 +168,7 @@ private[akka] final class ShardedDaemonProcessImpl(system: ActorSystem[_])
       messageClass: Class[T],
       name: String,
       numberOfInstances: Int,
-      behaviorFactory: function.Function[Int, Behavior[T]],
+      behaviorFactory: function.Function[Integer, Behavior[T]],
```

Comment:
by the way, we should use `java.util.function.IntFunction` instead

Reply:
> by the way, we should use java.util.function.IntFunction instead

Ok, I will change it on another PR